tpp-app/src/services/LogSymptomsService.js

```diff
@@ -52,21 +52,66 @@ export const POSTsymptomsForDate = async (day, month, year, symptoms) => new Pro
  * date.month is a number (January = 1),
  * date.year is a number.
  */
-export const LogMultipleDayPeriod = async (dates) => {
+export const LogMultipleDayPeriod = async (dates) => new Promise(async (resolve, reject) => {
     // run this code for each value in the dates array
-    dates.map(async (date)=>{
-        const year = date.year;
-        const month = date.month;
-        const day = date.day;
+    if(dates.length > 0){
+        try {
+            curYear = dates[0].year;
+            const calendarData = await getCalendarByYear(curYear);
+        
 
-        let symptoms = await GETsymptomsForDate(day, month, year)
+            dates.map((date)=>{
+                const year = date.year;
+                const month = date.month;
+                const day = date.day;
+                try {
+                    
+                    let symptoms = getSymptomsFromCalendar(calendarData, day, month, year);
 
-        if (symptoms.flow == null || symptoms.flow == FLOW_LEVEL.NONE){
-            symptoms.flow = FLOW_LEVEL.MEDIUM;
-        }
+                    if (symptoms.flow == null || symptoms.flow == FLOW_LEVEL.NONE){
+                        symptoms.flow = FLOW_LEVEL.MEDIUM;
+                    }
+
+                    calendarData[year][month-1][day-1] = symptoms;
+                } catch (error) {
+                    console.log(error);
+                }
+
+            })
+
+            if(calendarData[curYear]){
 
-        POSTsymptomsForDate(day, month, year, symptoms);
+                await AsyncStorage.setItem(curYear.toString(), JSON.stringify(calendarData[curYear]))
+                .then(() => resolve())
+                .catch((e) => {
+                    console.log(JSON.stringify(e));
+                    reject(`Unable to mergeItem and post symptoms for multiselect.`);
+                });
+                
+            }
+            
+            if(calendarData[curYear - 1]){
+                await AsyncStorage.setItem((curYear - 1).toString(), JSON.stringify(calendarData[curYear - 1]))
```

Comment:
I think it would be nicer to write a helper function with argument `year` & call it three times here, since there's repeated code

Reply:
I'll do this after we merge. No time rn :(

Reply:
sure. Just make sure @faye-tan knows so we add it as an issue on the todo issue

Reply:
Same comment a above, ok with leaving it for later, but @roylin2000 I'll assign you to go back in and make changes later then.